src/Compilers/VisualBasic/Portable/Symbols/Source/SourcePropertySymbol.vb

```diff
@@ -734,6 +734,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                                                                             isOverrides:=True, isWithEvents:=Me.IsWithEvents))
                 End If
 
+                Debug.Assert(IsDefinition)
```

Comment:
I am going to push this assert into OverriddenMember property. It will make it easier to avoid similar bugs in the future.